### ISSUE TITLE
Specifying Language with Internal Dependencies

### ISSUE THREAD

**Issue by phillipjohnston** [mesonbuild/meson#5723]
I have an internal dep which provides some compiler arguments. However, they should only be used with .cpp files and produce warning when applied to .c files.

Is there a good way for specifying a language with an internal dependency? I don't want to add to the global project options because you should only use the flags if you're consuming that particular dependency. If you set the flags and don't use the dependency, the build will fail.

```
libcxxabi_dep = declare_dependency(
	include_directories: include_directories('include/c++abi', is_system: true),
	link_with: libcxxabi,
	compile_args: stdlib_compiler_flags + cpp_stdlib_compiler_flags,
	link_args: stdlib_link_flags
)
```

Warning:

```
[656/1262] Compiling C object 'src/pro...rtos_src_portable_MemMang_heap_3.c.o'.
cc1: warning: command line option '-nostdinc++' is valid for C++/ObjC++ but not for C
```

**Comment by phillipjohnston** [comment]
`dependency()` has a `language` option, and build targets have `X_args` options, but `declare_dependency()` seems to throw both of those models to the wind :).

**Comment by phillipjohnston** [comment]
(Ping)

**Comment by phillipjohnston** [comment]
Closing due to lack of interest.

**Comment by dcbaker** [comment]
This is on my todo list, it's just a long todo list

**Comment by phillipjohnston** [comment]
Oh, that's great!

**Comment by dcbaker** [comment]
https://github.com/mesonbuild/meson/pull/14784 and https://github.com/mesonbuild/meson/pull/14199 are both groundwork on my way to do this, because this is currently one of my main interests in solving.

**Comment by ilg-ul** [comment]
Those two PRs were long merged, but apparently are not directly related to this issue.

Any plans to address this issue too?

**Comment by dcbaker** [comment]
Yes. I am still working toward fixing dependencies, I'd just like to fix all of the issues in one go, and the per-language keyword arguments are just one of the issues I see. I'll write up a more comprehensive issue
